lib/chef-api/resources/base.rb

```diff
@@ -927,7 +927,7 @@ def to_hash
     #
     # @return [String]
     #
-    def to_json
+    def to_json(state = nil)
```

Comment:
I think it expects one optional parameter, * allows more of them. Anyway, I changed it to what you suggested.